test/ce.integration.spec.ts

```diff
@@ -19,4 +20,91 @@ describe('ce.integration.spec.ts', () => {
+    class CustomElementTwo {
+      static $view = '<template>One</template>';
+    }
+    const { host, registry, dispose } = await bootstrapAurelia({
+      root: class RootViewModel {
+        static $view = '<template></template>';
+      },
+      resources: [CustomElementTwo]
+    });
+    registry.register(CustomElementTwo);
+    await waitForTimeout(50);
+
+    const customComponent = document.createElement("custom-element-two");
+    host.appendChild(customComponent);
+    expect(customComponent.textContent).toBe('One');
+    dispose();
+  });
+
+  it('should not initialise web component when attribute changed on disconnected element', async () => {
+    class CustomElementThree {
+      static $view = '<template>One</template>';
+      // Bindable property must exist in order for the attributeChangedCallback to be executed
+      @bindable testing
+    }
+    const { registry, dispose } = await bootstrapAurelia({
+      root: class RootViewModel {
+        static $view = '<template></template>';
+      },
+      resources: [CustomElementThree]
+    });
+    registry.register(CustomElementThree);
+    await waitForTimeout(50);
+
+    const customComponent = document.createElement("custom-element-three");
+    customComponent.setAttribute("testing", "xx-ignored-xx");
+    expect(customComponent.textContent).toBe('One');
+    dispose();
+  });
+
+  it('should initialise web component with modified attribute value', async () => {
+    class CustomElementFour {
+      static $view = '<template>${testing}</template>';
+      // Bindable property must exist in order for the attributeChangedCallback to be executed
+      @bindable testing
+    }
+    const { host, registry, dispose } = await bootstrapAurelia({
+      root: class RootViewModel {
+        static $view = '<template></template>';
+      },
+      resources: [CustomElementFour]
+    });
+    registry.register(CustomElementFour);
+    await waitForTimeout(50);
+
+    const customComponent = document.createElement("custom-element-four");
+    customComponent.setAttribute("testing", "One");
+    host.appendChild(customComponent);
+    expect(customComponent.textContent).toBe('One');
+    dispose();
+  });
+
+  it('should update web component when attribute value is modified later', async () => {
+    class CustomElementFive {
+      static $view = '<template>${testing}</template>';
+      // Bindable property must exist in order for the attributeChangedCallback to be executed
+      @bindable testing
+    }
+    const { host, registry, dispose } = await bootstrapAurelia({
+      root: class RootViewModel {
+        static $view = '<template></template>';
+      },
+      resources: [CustomElementFive]
+    });
+    registry.register(CustomElementFive);
+    await waitForTimeout(50);
+
+    const customComponent = document.createElement("custom-element-five");
+    host.appendChild(customComponent);
+    await waitForTimeout(50);
+
+    customComponent.setAttribute("testing", "One");
```

Comment:
lets add an assertion here that text content should not be `'One'`

Reply:
I think doing that, it would help communicate the timing clearer

Reply:
Done.